QuizApp/App/Screens/Quiz/QuizViewController.swift

```diff
@@ -15,6 +26,70 @@ final class QuizViewController: UIViewController {
         createViews()
         styleViews()
         defineLayoutForViews()
+        setupDelegateAndDataSource()
+    }
+
+}
+
+// MARK: - UICollectionViewDataSource methods
+extension QuizViewController: UICollectionViewDataSource {
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        if collectionView == categoryCollectionView {
+            return quizViewModel.quizCategories.count
+        } else {
+            return 4
+        }
+    }
+
+    func collectionView(
+        _ collectionView: UICollectionView,
+        cellForItemAt indexPath: IndexPath
+    ) -> UICollectionViewCell {
+        if collectionView == categoryCollectionView {
+            guard let cell = collectionView.dequeueReusableCell(
+                withReuseIdentifier: CategoryCell.reuseIdentifier,
+                for: indexPath
+            ) as? CategoryCell else { fatalError() }
+
+            cell.data = Category(name: quizViewModel.quizCategories[indexPath.item].name)
+
+            if indexPath.item == 0 {
+                collectionView.selectItem(at: indexPath, animated: true, scrollPosition: .left)
+            }
+
+            return cell
+        } else {
+            guard let cell = collectionView.dequeueReusableCell(
+                withReuseIdentifier: QuizCell.reuseIdentifier,
+                for: indexPath
+            ) as? QuizCell else { fatalError() }
+
+            cell.data = Category(name: quizViewModel.quizCategories[indexPath.item].name)
+            return cell
+        }
+    }
+
+}
+
+// MARK: - UICollectionViewDelegateFlowLayout methods
+extension QuizViewController: UICollectionViewDelegateFlowLayout {
+    func collectionView(
+        _ collectionView: UICollectionView,
+        layout collectionViewLayout: UICollectionViewLayout,
+        sizeForItemAt indexPath: IndexPath
+    ) -> CGSize {
+        if collectionView == categoryCollectionView {
+            return calculateSize(indexPath: indexPath)
```

Comment:
Use `estimatedItemSize` property of UICollectionViewFlowDelegate to create auto-sizing cells.